lib/instrumentation/generic-span.js

```diff
@@ -101,3 +120,16 @@ GenericSpan.prototype.addLabels = function (labels) {
   }
   return true
 }
+
+GenericSpan.prototype.setType = function (type, subtype, action) {
+  if (type && type.indexOf('.') >= 0) {
+    const parts = type.split('.')
+    type = parts.shift()
+    subtype = parts.shift()
+    action = parts.shift()
+  }
+
+  this._type = type || 'custom'
```

Comment:
I think we can probably drop the string "custom" as a fallback in this refactor as well. If I recall correctly, the APM UI will see `null` and group transactions with `null` under its own group named "Custom" in the UI. And spans with type `null` will get special treatment as well. Right @elastic/apm-ui?

Reply:
fixed

Reply:
Did you check that the UI behaves as I think? I'm not 100%

Reply:
We currently don't group transactions with `name` property. We did talk about it but never got around to it. 
Is this also something that should happen for transactions without `type`? 

>  And spans with type null

Spans are only displayed in the timeline view. Afair we show spans regardless of their type - the type is only used to display an icon.

Reply:
My bad... spans and transactions are always required to have a type property